docs/src/content/docs/concepts/arrays.mdx

```diff
@@ -0,0 +1,202 @@
+
+const documents = await queries
+  .flatMap((query) => (
+    <WebSearch query={query} /> // returns a list of documents
+  ))
+  .toArray();
+```
+
+### Filter
+
+Filters allow you to filter out elements that don't match a given condition. Inside the filter predicate, you can pass either:
+
+- A component that returns a boolean
+- A plain boolean expression
+
+#### Filtering with a component
+
+When you filter with a component, the component needs to result in a boolean value.
+
+```tsx
+// Define a component that returns a boolean
+const EvenNumberFilter = gsx.Component<number, boolean>(
+  "EvenNumberFilter",
+  async ({ value }) => {
+    return value % 2 === 0;
+  },
+);
+
+// Use a component that returns a boolean to filter
+const evenNumbers = await gsx
+  .array([10, 11, 12, 13, 14])
+  .filter((n) => <EvenNumberFilter value={n} />)
+  .toArray();
+```
+
+You can also use the component's child function to convert the output of a component into a boolean:
+
+```tsx
+const evenNumbers = await gsx
+  .array([10, 11, 12, 13, 14])
+  .filter((n) => (
+    <EvenOrOdd value={n}>
+      {({ result: string }) => result.toLowerCase() === "even"}
+    </EvenOrOdd>
+  ))
+  .toArray();
+```
+
+#### Filter using a plain boolean expression
+
+When you filter with a boolean expression, you use the filter method
+
+```tsx
+const evenNumbers = await gsx
+  .array([10, 11, 12, 13, 14])
+  .filter((n) => n % 2 === 0)
+  .toArray();
+```
+
+#### Filter using index and array parameters
+
+Filter expressions can also access the current index and the entire array. This is useful for scenarios where you need to filter based on the position of an element in the array or find unique values.
+
+Here's a basic example of using this to remove duplicates from an array:
+
+```tsx
+const uniqueNumbers = await gsx
+  .array([1, 2, 2, 3, 3, 3, 4])
+  .filter((num, index, array) => array.indexOf(num) === index)
+  .toArray();
+// Result: [1, 2, 3, 4]
+```
+
+### Reduce
+
+The reduce operation allows you to process all of the items in an array to produce a single value.
+
+For example, if you wanted to translate a markdown document section by section and combine the results into a single document, it would look something like this:
+
+```tsx
+const markdownContent = "<some markdown content>";
+
+const translatedContent = await gsx
+  .array(markdownContent.split(/(?=^#{1,2} )/m))
+  .map((value) => <TranslateSection value={value} />)
+  .reduce((acc, value) => <CombineSections acc={acc} value={value} />, "");
+```
+
+To break down this example a bit more:
+
+- `acc` is the accumulator, which starts as the initial value, in this case an empty string
+- `value` is the current section of the markdown content
+- `CombineSections` is a component that does the combining given both the accumulator and the current value
+
+## How array operations work
+
+When you boil it down, `gsx.array` is essentially a wrapper around the `gsx.execute` function.
```

Comment:
I'd remove the "boil you it down". Maybe just: 

`gsx.array` is a conveinece wrapper around `gsx.execute` and the corresponding javascript array function. 

From there you could show the same `.map` code written via gsx.array and using gsx.execute to demonstrate.